rpc-rs/src/provider_main.rs

```diff
@@ -290,90 +295,164 @@ where
+fn configure_tracing(
+    provider_name: String,
+    structured_logging_enabled: bool,
+    log_level_override: Option<String>,
+) {
+    let base_reg = tracing_subscriber::Registry::default();
+    let level_filter = get_level_filter(log_level_override);
+
+    let maybe_tracer = (std::env::var_os("OTEL_TRACES_EXPORTER")
         .unwrap_or_default()
         .to_ascii_lowercase()
-        == "otlp"
-    {
-        let mut tracing_endpoint = std::env::var("OTEL_EXPORTER_OTLP_ENDPOINT")
-            .unwrap_or_else(|_| format!("http://localhost:55681{}", TRACING_PATH));
-        if !tracing_endpoint.ends_with(TRACING_PATH) {
-            tracing_endpoint.push_str(TRACING_PATH);
+        == "otlp")
+        .then(|| get_tracer(provider_name));
+
+    let res = match (maybe_tracer, structured_logging_enabled) {
+        (Some(Ok(t)), true) => {
+            let log_layer = get_json_log_layer();
+            let tracing_layer = tracing_opentelemetry::layer().with_tracer(t);
+            let layered = base_reg.with(level_filter).with(log_layer).with(tracing_layer);
+            tracing::subscriber::set_global_default(layered)
         }
-        match opentelemetry_otlp::new_pipeline()
-            .tracing()
-            .with_exporter(
-                opentelemetry_otlp::new_exporter()
-                    .http()
-                    .with_endpoint(tracing_endpoint)
-                    .with_protocol(Protocol::HttpBinary),
-            )
-            .with_trace_config(
-                trace::config()
-                    .with_sampler(Sampler::AlwaysOn)
-                    .with_id_generator(IdGenerator::default())
-                    .with_max_events_per_span(64)
-                    .with_max_attributes_per_span(16)
-                    .with_max_events_per_span(16)
-                    .with_resource(Resource::new(vec![opentelemetry::KeyValue::new(
-                        "service.name",
-                        provider_name,
-                    )])),
-            )
-            .install_batch(opentelemetry::runtime::Tokio)
-        {
-            Ok(t) => tracing::subscriber::set_global_default(
-                subscriber.with(tracing_opentelemetry::layer().with_tracer(t)),
-            ),
-            Err(e) => {
-                eprintln!(
-                    "Unable to configure OTEL tracing, defaulting to logging only: {:?}",
-                    e
-                );
-                tracing::subscriber::set_global_default(subscriber)
-            }
+        (Some(Ok(t)), false) => {
+            let log_layer = get_default_log_layer();
+            let tracing_layer = tracing_opentelemetry::layer().with_tracer(t);
+            let layered = base_reg.with(level_filter).with(log_layer).with(tracing_layer);
+            tracing::subscriber::set_global_default(layered)
+        }
+        (Some(Err(err)), true) => {
+            eprintln!("Unable to configure OTEL tracing, defaulting to logging only: {err:?}");
```

Comment:
Is this an error if we don't supply OTEL parameters or should it just fail silently?